### PR TITLE
Changes for getting youtube feeds

### PR DESCRIPTION
**Pull request by shamnadps**

### CODE REVIEW

File: backend/src/youtube-feed/youtube-feed.service.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import axios from 'axios';
+
+const config = new ConfigService();
+
+const API_KEY = config.youTubeApiKey;
+const CHANNEL_ID = config.youTubeChannelID;
+const MAX_RESULTS = config.youTubeChannelListLimit;
+
+const youtube_fetch_url = `https://www.googleapis.com/youtube/v3/search?`
+    + `key=${API_KEY}`
```
Comment (StormBlade97) [inline backend/src/youtube-feed/youtube-feed.service.ts]: you are already using tagged template expression `the \`` It support multiline.
```
`You can write
 code like this
 and ${inject} thing just fine
 `
```

File: backend/src/youtube-feed/youtube-feed.service.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    async onModuleInit() {
+        await this.fetchAndSaveYouTubeFeed();
+    }
+
+    @Transaction()
+    async fetchAndSaveYouTubeFeed() {
+        try {
+            const feed = await axios.get(youtube_fetch_url);
+            const videoIds = feed.data.items.map(item => item.id.videoId);
+            const videoDetails = await axios.get(youtube_video_details_url + videoIds.filter(id => id));
```
Comment (StormBlade97) [inline backend/src/youtube-feed/youtube-feed.service.ts]: You dont have to do this, but for quality of code, you should have smaller try/catch phrase to have slightly more detailed errors.

File: backend/src/youtube-feed/youtube-feed.service.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            const youTubeFeeds = feed.data.items.map(item => {
```
Comment (StormBlade97) [inline backend/src/youtube-feed/youtube-feed.service.ts]: Use promise.all here

File: backend/src/youtube-feed/youtube-feed.service.ts (hunk 2 above)
Comment (StormBlade97) [inline backend/src/youtube-feed/youtube-feed.service.ts]: is this detail of 1 video, or multiple videos? If multiple videos then why only 1 axios call? This looks like a bug

File: backend/src/youtube-feed/youtube-feed.controller.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        private readonly logger: Logger) {
+        this.logger = new Logger('YouTubeFeedController');
+    }
+
+    @Get()
+    async findAll(@Res() response): Promise<YouTubeFeed[]> {
+        try {
+            const youTubeFeeds = await this.youTubeFeedService.findAll();
+            if (youTubeFeeds) {
+                return response.status(200).json(youTubeFeeds);
```
[the remark this answers is not in the document]
Comment (shamnadps) [reply backend/src/youtube-feed/youtube-feed.controller.ts]: yes will do that :)

File: backend/src/youtube-feed/youtube-feed.service.ts (hunk 3 above)
Comment (shamnadps) [reply backend/src/youtube-feed/youtube-feed.service.ts]: why do we need this there. are not async calls inside the map.

File: backend/src/youtube-feed/youtube-feed.service.ts (hunk 2 above)
Comment (shamnadps) [reply backend/src/youtube-feed/youtube-feed.service.ts]: its for multiple videos. the api is like that from youtube.

File: backend/src/youtube-feed/youtube-feed.service.ts (hunk 1 above)
Comment (shamnadps) [reply backend/src/youtube-feed/youtube-feed.service.ts]: yes thank you

**Pull request merged**